api/stats.proto

```diff
@@ -54,73 +54,84 @@ message StatsConfig {
   //
   // .. note::
   //
-  //   If any default tags are specified twice, the config will be considered
+  //   If any tag names are specified twice, the config will be considered
   //   invalid.
   string tag_name = 1;
 
-  // The first capture group identifies the portion of the name to remove. The
-  // second capture group (which will normally be nested inside the first) will
-  // designate the value of the tag for the statistic. If no second capture
-  // group is provided, the first will also be used to set the value of the tag.
-  // All other capture groups will be ignored.
-  //
-  // Take for example, with a stat name ``cluster.foo_cluster.upstream_rq_timeout``
-  // and
-  //
-  // .. code-block:: json
-  //
-  //   {
-  //     "tag_name": "envoy.cluster_name",
-  //     "regex": "^cluster\.((.+?)\.)"
-  //   }
-  //
-  // Note that the regex will remove ``foo_cluster.`` making the tag extracted
-  // name ``cluster.upstream_rq_timeout`` and the tag value for
-  // ``envoy.cluster_name`` will be ``foo_cluster`` (note: there will be no
-  // ``.`` character because of the second capture group).
-  //
-  // An example with two regexes and stat name
-  // ``http.connection_manager_1.user_agent.ios.downstream_cx_total``:
-  //
-  // .. code-block:: json
-  //
-  //   [
-  //     {
-  //       "tag_name": "envoy.http_user_agent",
-  //       "regex": "^http(?=\.).*?\.user_agent\.((.+?)\.)\w+?$"
-  //     },
-  //     {
-  //       "tag_name": "envoy.http_conn_manager_prefix",
-  //       "regex": "^http\.((.*?)\.)"
-  //     }
-  //   ]
-  //
-  // The first regex will remove ``ios.``, leaving the tag extracted name
-  // ``http.connection_manager_1.user_agent.downstream_cx_total``. The tag
-  // ``envoy.http_user_agent`` will be added with tag value ``ios``.
-  //
-  // The second regex will remove ``connection_manager_1.`` from the tag
-  // extracted name produced by the first regex
-  // ``http.connection_manager_1.user_agent.downstream_cx_total``, leaving
-  // ``http.user_agent.downstream_cx_total`` as the tag extracted name. The tag
-  // ``envoy.http_conn_manager_prefix`` will be added with the tag value
-  // ``connection_manager_1``.
-  string regex = 2;
+  oneof tag_value {
+    // Designates a tag to strip from the tag extracted name and provide as a named
+    // tag value for all statistics. This will only occur if any part of the name
+    // matches the regex provided with one or more capture groups.
+    //
+    // The first capture group identifies the portion of the name to remove. The
+    // second capture group (which will normally be nested inside the first) will
+    // designate the value of the tag for the statistic. If no second capture
+    // group is provided, the first will also be used to set the value of the tag.
+    // All other capture groups will be ignored.
+    //
+    // Take for example, with a stat name ``cluster.foo_cluster.upstream_rq_timeout``
```

Comment:
How about format like this:

Example 1: xxx

description

code

Reply:
+1. The structured example format will help readers.

I think the code is placed there as a example condition description, we need to keep the code above the description. So changed like 8631e4f7ee0c0b10c756ac0307e590cfc4eda62a.